lib/helpers.bash

```diff
@@ -92,980 +96,946 @@ alias reload_completion="$(_make_reload_alias completion completion)"
-  if _bash-it-profile-load-parse-profile "$profile_path" "dry"; then
-    echo "Profile \"$name\" parsed successfully!"
-    echo "Disabling current configuration..."
-    _disable-all
-    echo ""
-    echo "Enabling configuration based on profile..."
-    _bash-it-profile-load-parse-profile "$profile_path"
-    echo ""
-    echo "Profile \"$name\" enabled!"
-  fi
-}
+	_about 'loads a configuration from the "profiles" directory'
+	_group 'lib'
+
+	local name="$1"
+	if [[ -z $name ]]; then
+		echo -e "${echo_orange?}Please specify profile name to load, not changing configuration...${echo_reset_color?}"
+		return 1
+	fi
+
+	local profile_path="${BASH_IT}/profiles/$name.bash_it"
+	if [[ ! -f "$profile_path" ]]; then
+		echo -e "${echo_orange?}Could not find profile '$name', not changing configuration...${echo_reset_color?}"
+		return 1
+	fi
 
-_bash-it-restart() {
-  _about 'restarts the shell in order to fully reload it'
-  _group 'lib'
-
-  saved_pwd="${PWD}"
-
-  case $OSTYPE in
-    darwin*)
-      init_file=.bash_profile
-      ;;
-    *)
-      init_file=.bashrc
-      ;;
-  esac
-  exec "${0/-/}" --rcfile <(echo "source \"$HOME/$init_file\"; cd \"$saved_pwd\"")
+	echo "Trying to parse profile '$name'..."
+	if _bash-it-profile-load-parse-profile "$profile_path" "dry"; then
+		echo "Profile '$name' parsed successfully!"
+		echo "Disabling current configuration..."
+		_disable-all
+		echo ""
+		echo "Enabling configuration based on profile..."
+		_bash-it-profile-load-parse-profile "$profile_path"
+		echo ""
+		echo "Profile '$name' enabled!"
+	else
+		false # failure
+	fi
 }
 
-_bash-it-reload() {
-  _about 'reloads a profile file'
-  _group 'lib'
+function _bash-it-restart() {
+	_about 'restarts the shell in order to fully reload it'
+	_group 'lib'
+
+	local saved_pwd="${PWD}" init_file
+
+	if shopt -q login_shell; then
+		init_file=.bash_profile
+	else
+		init_file=.bashrc
+	fi
+	exec "${0/-/}" --rcfile <(echo "source \"$HOME/$init_file\"; cd \"$saved_pwd\"")
+}
 
-  pushd "${BASH_IT}" &> /dev/null || return
+function _bash-it-reload() {
+	_about 'reloads a profile file'
+	_group 'lib'
 
-  case $OSTYPE in
-    darwin*)
-      source ~/.bash_profile
-      ;;
-    *)
-      source ~/.bashrc
-      ;;
-  esac
+	pushd "${BASH_IT?}" > /dev/null || return
 
-  popd &> /dev/null || return
+	# shellcheck disable=SC1090
+	if shopt -q login_shell; then
+		# shellcheck source-path=$HOME
+		source ~/.bash_profile
+	else
+		# shellcheck source-path=$HOME
+		source ~/.bashrc
+	fi
+	popd > /dev/null || return
 }
 
-_bash-it-determine-component-status-from-path ()
-{
-  _about 'internal function used to process component name and check if its enabled'
-  _param '1: full path to available component file'
-  _example '$ _bash-it-determine-component-status-from-path "${BASH_IT}/plugins/available/git.plugin.bash'
-
-  # Check for both the old format without the load priority, and the extended format with the priority
-  declare enabled_files enabled_file
```

Comment:
@NoahGorny, I recommend using `local` basically always instead of `declare`. It's not obvious that `declare` creates local variables, but it always does (except v5+ with `declare -g`).